nvflare/lighter/impl/cert.py

```diff
@@ -35,47 +105,62 @@ def __init__(self):
         information about previously generated certs, it loads them back and reuses them.
         """
         self.root_cert = None
-        self.persistent_state = dict()
+        self.persistent_state = None
+        self.serialized_cert = None
+        self.pri_key = None
+        self.pub_key = None
+        self.subject = None
+        self.issuer = None
 
-    def initialize(self, ctx):
-        state_dir = self.get_state_dir(ctx)
-        cert_file = os.path.join(state_dir, "cert.json")
-        if os.path.exists(cert_file):
-            self.persistent_state = json.load(open(cert_file, "rt"))
-            self.serialized_cert = self.persistent_state["root_cert"].encode("ascii")
+    def initialize(self, project: Project, ctx: ProvisionContext):
```

Comment:
When creating the ProvisionContext(), the project is already placed into the ProvisionContext instance. No need to pass in the "project" parameter any more. Better get it from the ProvisionContext. Also the concept is clearer.